### PR TITLE
[Enhancement] publish version return tablet even though it's transaction have finished

### LINKED ISSUE #16370: [Enhancement] publish version return tablet even though it's transaction have finished
## Enhancement
If FE miss the last finish_publish_version req return by BE, and then retry send publish_version req to BE, BE wouldn't return the tablet's version back to FE if transaction has finished. So FE have to wait for tablet infos report by BE, and then this transaction can be visible. Reliable on tablet infos report by BE to make transaction visible can be very slow.

So we can let BE returns the tablet's version back to FE if transaction has finished, to make transaction be visible more quickly.

### PR DESCRIPTION
**Pull request by luohaha**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16370

## Problem Summary(Required) ：
If FE miss the last finish_publish_version req return by BE, and then retry send publish_version req to BE, BE wouldn't return the tablet's version back to FE if transaction has finished. So FE have to wait for tablet infos report by BE, and then this transaction can be visible. Reliable on tablet infos report by BE to make transaction visible can be very slow.

So we can let BE returns the tablet's version back to FE if transaction has finished, to make transaction be visible more quickly.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionDesc.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|1|1|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionDesc.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|1|1|100.00%|[]|

**Comment by luohaha** [comment]
run starrocks_be_unittest

**Comment by luohaha** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionDesc.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|1|1|100.00%|[]|

**Review by decster — DISMISSED** [review]

File: be/src/agent/publish_version.cpp [hunk 1]
```diff
@@ -163,6 +163,34 @@ void run_publish_version_task(ThreadPoolToken* token, const TPublishVersionReque
             pair.__set_version(task.max_continuous_version);
         }
     }
+    // return tablet and it's version which have already finished.
```
Comment (nshangyiming) [inline be/src/agent/publish_version.cpp:166]:
```suggestion
    // return tablet and its version which has already finished.
```

**Review by nshangyiming — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/agent/publish_version.cpp (hunk 1 above)
Comment (meegoo) [inline be/src/agent/publish_version.cpp:166]: It seems a little unclear to change the whole process in this way. The version will be set through both the task and the tablet. Can it be directly changed to set the version according to the tablet?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/RangePartitionDesc.java|0|1|00.00%|[84]|
|:large_blue_circle:|com/starrocks/sql/ast/ListPartitionDesc.java|1|1|100.00%|[]|

File: be/src/agent/publish_version.cpp (hunk 1 above)
Comment (luohaha) [reply be/src/agent/publish_version.cpp]: Done

**Review by decster — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by luohaha** [comment]
run starrocks_admit_test

**Comment by luohaha** [comment]
https://github.com/Mergifyio backport branch-2.5

**Pull request merged**